### DISCUSSION TITLE
Using Parameter in ShellTask command

### DISCUSSION THREAD

**Discussion by NoName115** [PrefectHQ/prefect#4200] [Q&A] [answered]
Hi,

Is there any way how I can use an Parameter task which will define how the command for ShellTask will looks like.
Let's take an example:
```python

sl = ShellTask()

with Flow(name="test") as fl:
  ls_args = Parameter("ls_args", required=True)
  sl(command="ls {ls_args}")
```

But this approach will set Parameter object to a command argument, is there a way to do this?
Thanks a lot for your answer.

**Comment by cicdw** [comment, marked as answer]
Hi @NoName115 - yes this is possible; remember that Flows represent _deferred_ computations that run in the future, not now.  Therefore any logic that you want to include runtime values (e.g., parameters) need to be encapsulated in a task or one of the hooks Prefect provides (e.g., state handlers).  In your case, Python is formatting the string _now_ instead of when the flow runs.

All of that being said, a simple new task will do the trick:

```python
sl = ShellTask()

@task
def format_command(ls_args):
    return f"ls {ls_args}"

with Flow(name="test") as fl:
  ls_args = Parameter("ls_args", required=True)
  sl(command=format_command(ls_args))
```

**Reply by NoName115** [reply]
Hi @cicdw thanks a lot for your answer, I wasn't thinking that I can pass another Task as an argument to command :)
